packages/@aws-cdk/aws-synthetics/lib/test-script.ts

```diff
@@ -0,0 +1,94 @@
+import * as s3 from '@aws-cdk/aws-s3';
+import { Construct } from '@aws-cdk/core';
+import { Code } from './code';
+
+/**
+ * Specify a test that the canary should run
+ */
+export class Test {
+  /**
+   * Specify a custom test with your own code
+   *
+   * @returns `Test` associated with the specified Code object
+   * @param code The script you want the canary to run
+   * @param handler The handler of the code
+   */
+  public static custom(scope: Construct, options: CustomOptions): Test {
+    const codeConfig = options.code.bind(scope);
```

Comment:
I understand this is why you accept a `scope`, but actually the binding should not take place here. It should be done in the constructor of the canary, avoiding the user from having to pass it.

For example, in the canary constructor:

```typescript
const codeConfig = props.code.bind(this)
```

Notice how `this` is used as the scope, since you are inside the canary construct. 

See https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-lambda/lib/function.ts#L535